xarray/tests/test_backends.py

```diff
@@ -2020,6 +2034,9 @@ def test_cmp_local_file(self):
             self.assertNotIn('NC_GLOBAL', actual.attrs)
             self.assertIn('history', actual.attrs)
 
+            # attributes and encoding
+            assert actual.attrs.keys() == expected.attrs.keys()
```

Comment:
In an ideal world, we would use `assertDatasetIdentical()` to also check attributes exactly.

However, looking at the bears example as served by the test DAP server, it is not clear to me what the relation is between the original netCDF file and what the Hyrax DAP server serves up. It seems to insert some extra attributes. I'm not sure it's possible to check all of these exactly without encoding details of Hyrax in xarray's test suite.

This seems like about the best we can do for now. But can you add a comment about this? Something like:
```
# we don't check attributes exactly with assertDatasetIdentical() because
# the test DAP server seems to insert some extra attributes not found in the
# netCDF file.
```